tests/schema/fail/invalid-components.yaml

```diff
@@ -0,0 +1,163 @@
+info:
+  title: API
+  version: 1.0.0
+components:
+  pathItems:
+    ö: true                       # wrong map key pattern
+    invalid:
+      $ref: 42                    # must be a string
+      summary: true               # must be a string
+      description: true           # must be a string
+      servers: true               # must be an array
+      parameters: true            # must be an array
+    invalid-operations:
+      get:
+        tags: true                # must be an array
+        summary: true             # must be a string
+        description: true         # must be a string
+        operationId: true         # must be a string
+        parameters: true          # must be an array
+        callbacks: true           # must be an object
+        deprecated: maybe         # must be a boolean
+        security: no              # must be an array
+        servers: none             # must be an array
+      patch: false                # must be an object
+    invalid-responses:
+      get:
+        tags: [true]              # array items must be strings
+        responses: false          # must be an object
+      patch:
+        responses: {}             # must have at least one field
+      post:
+        responses:
+          invalid: true           # must be an object
+  requestBodies:
+    no-object: true      # must be an object
+    invalid:
+      description: true  # must be a string
+      required: no       # must be a boolean
+  parameters:
+    no-object: true      # must be an object
+    no-in:               # object must have `in` field
+      name: id
+      schema:
+        type: string
+    invalid:
+      name: 42            # must be a string
+      in: invalid         # must be one of `query`, `header`, `path`, `cookie`
+      description: true   # must be a string
+      required: no        # must be a boolean
+      deprecated: no      # must be a boolean
+      content: {}         # must have at least one field
+    invalid-content:
+      name: id
+      content: 
+        one: true         # must be an object
+        two: false        # must be an object
+    invalid-querystring:  # must have `content` field
+      name: id
+      in: querystring     
+    optional-path:
+      name: id
+      in: path
+      schema:
+        type: string
+      required: false     # must be true for path parameters
+    invalid-path:
+      name: id
+      in: path
+      schema:
+        type: string
+      style: invalid      # must be one of the allowed enum values
+      allowReserved: yes  # must be a boolean
+    invalid-header:
+      name: id
+      in: header
+      schema:
+        type: string
+      style: 42           # must be the string "simple"
+    invalid-query:
+      name: id
+      in: query
+      schema:
+        type: string
+      style: invalid        # must be one of the allowed enum values
+      allowEmptyValue: yes  # must be a boolean
+      allowReserved: no     # must be a boolean
```

Comment:
Are these supposed to be testing that they are _not_ booleans?  because `yes` and `no` are booleans in yaml:

```pycon
Python 3.10.10 (main, Feb 16 2023, 23:23:55) [Clang 14.0.0 (clang-1400.0.29.202)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> from yaml import safe_load as sl
>>> sl("""
... foo: yes
... bar: no
... """)
{'foo': True, 'bar': False}
>>> 
```

Is the coverage tool showing this testing the negative case for these fields?